huaweicloud/services/accessanalyzer/data_source_huaweicloud_access_analyzers.go

```diff
@@ -0,0 +1,246 @@
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"unused_access": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"unused_access_age": {
+													Type:     schema.TypeInt,
+													Computed: true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						"tags": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status_reason": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"code": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"details": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"urn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"organization_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"last_analyzed_resource": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"last_resource_analyzed_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"created_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func buildListAccessAnalyzerParams(d *schema.ResourceData, marker string) string {
+	res := ""
+	if analyzerType, ok := d.GetOk("type"); ok {
+		res = fmt.Sprintf("%s&type=%v", res, analyzerType)
+	}
+
+	if marker != "" {
+		res = fmt.Sprintf("%s&marker=%v", res, marker)
+	}
+
+	if res != "" {
+		res = "?" + res[1:]
+	}
+
+	return res
+}
+
+func queryAccessAnalyzer(client *golangsdk.ServiceClient, d *schema.ResourceData) ([]interface{}, error) {
+	var (
+		httpUrl = "v5/analyzers"
+		result  = make([]interface{}, 0)
+	)
+
+	listPath := client.Endpoint + httpUrl
+
+	queryParams := buildListAccessAnalyzerParams(d, "")
+	listPath += queryParams
+
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	for {
+		requestResp, err := client.Request("GET", listPath, &opt)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving access analyzers: %s", err)
+		}
+		respBody, err := utils.FlattenResponse(requestResp)
+		if err != nil {
+			return nil, err
+		}
+		analyzers := utils.PathSearch("analyzers", respBody, make([]interface{}, 0)).([]interface{})
+		result = append(result, analyzers...)
+
+		marker := utils.PathSearch("page_info.next_marker", respBody, "").(string)
+		if marker == "" {
+			break
+		}
+
+		queryParams = buildListAccessAnalyzerParams(d, marker)
+		listPath += queryParams
```

Comment:
the result is wrong and it will contain many marker

Reply:
fixed